File: SolaceCore/Sources/SolaceCore/Models/Appointment.swift
import Foundation

public enum AppointmentStatus: String, Codable, Sendable, CaseIterable, Equatable {
    case pending
    case confirmed
    case declined
    case cancelled
}

public struct Appointment: Identifiable, Codable, Sendable, Equatable, Hashable {
    public let id: String
    public var studentID: String
    public var studentName: String
    public var counselorID: String
    public var counselorName: String
    public var scheduledAt: Date
    public var status: AppointmentStatus
    public var createdAt: Date

    public init(
        id: String,
        studentID: String,
        studentName: String,
        counselorID: String,
        counselorName: String,
        scheduledAt: Date,
        status: AppointmentStatus = .pending,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.studentID = studentID
        self.studentName = studentName
        self.counselorID = counselorID
        self.counselorName = counselorName
        self.scheduledAt = scheduledAt
        self.status = status
        self.createdAt = createdAt
    }

    public var participantIDs: [String] { [studentID, counselorID] }

    public func otherParticipantName(currentUserID: String) -> String {
        currentUserID == studentID ? counselorName : studentName
    }
}

File: SolaceAndroid/app/src/main/java/com/jonathanalumasa/solace/model/Appointment.kt
package com.jonathanalumasa.solace.model

import java.util.Date

/** Mirrors `SolaceCore.AppointmentStatus`. */
enum class AppointmentStatus(val rawValue: String, val label: String) {
    PENDING("pending", "Pending"),
    CONFIRMED("confirmed", "Confirmed"),
    DECLINED("declined", "Declined"),
    CANCELLED("cancelled", "Cancelled");

    companion object {
        fun fromRaw(raw: String?): AppointmentStatus? = entries.firstOrNull { it.rawValue == raw }
    }
}

/** Mirrors `SolaceCore.Appointment`. */
data class Appointment(
    val id: String,
    val studentID: String,
    val studentName: String,
    val counselorID: String,
    val counselorName: String,
    val scheduledAt: Date,
    val status: AppointmentStatus = AppointmentStatus.PENDING,
    val createdAt: Date = Date()
) {
    val participantIDs: List<String> get() = listOf(studentID, counselorID)

    fun otherParticipantName(currentUserID: String): String =
        if (currentUserID == studentID) counselorName else studentName
}
